src/main/java/edu/kit/provideq/toolbox/api/SolversRouter.java

```
@@ -52,29 +56,54 @@ private RouterFunction<ServerResponse> defineSolversRouteForMetaSolver(
-                .content(contentBuilder()
-                    .mediaType(APPLICATION_JSON_VALUE)
-                    .array(arraySchemaBuilder().schema(
-                        schemaBuilder().implementation(ProblemSolverInfo.class)))
-                )
-            )
+        ops -> handleSolversRouteDocumentation(ops, metaSolver)
     ).build();
   }
 
   private Mono<ServerResponse> handleSolversRouteForMetaSolver(MetaSolver<?, ?, ?> metaSolver) {
-    var solvers = metaSolver.getAllSolvers().stream()
-        .map(solver -> new ProblemSolverInfo(solver.getId(), solver.getName()))
-        .toList();
+    var solvers = getAllSolverInfos(metaSolver);
 
     return ok().body(Mono.just(solvers), new ParameterizedTypeReference<>() {
     });
   }
 
+  private static List<ProblemSolverInfo> getAllSolverInfos(MetaSolver<?, ?, ?> metaSolver) {
+    return metaSolver.getAllSolvers().stream()
+            .map(solver -> new ProblemSolverInfo(solver.getId(), solver.getName()))
+            .toList();
+  }
+
+  private void handleSolversRouteDocumentation(Builder ops, MetaSolver<?, ?, ?> metaSolver) {
+    ops
+        .operationId(getSolversRouteForProblemType(metaSolver.getProblemType()))
+        .tag(metaSolver.getProblemType().getId())
+        .description("Returns a list of solvers available to solve the "
+            + metaSolver.getProblemType().getId() + " problem type.")
+        .response(responseBuilder()
+            .responseCode(String.valueOf(HttpStatus.OK.value()))
+            .content(getOkResponseContent(metaSolver))
+        );
+  }
+
+  private static org.springdoc.core.fn.builders.content.Builder getOkResponseContent(
+          MetaSolver<?, ?, ?> metaSolver) {
+    var allSolvers = getAllSolverInfos(metaSolver);
+    String example;
+    try {
+      example = new ObjectMapper().writeValueAsString(allSolvers);
+    } catch (JsonProcessingException e) {
+      example = "Error: solvers could not be parsed";
```

please throw an exception instead